src/dds/rs-dds-device-proxy.cpp

```diff
@@ -261,6 +277,76 @@ dds_device_proxy::dds_device_proxy( std::shared_ptr< context > ctx, std::shared_
 }
 
 
+int dds_device_proxy::get_index_from_stream_name( const std::string & name ) const
+{
+    int index = 0;
+
+    auto delimiter_pos = name.find( '_' );
+    if( delimiter_pos != std::string::npos )
+    {
+        std::string index_as_string = name.substr( delimiter_pos + 1, name.size() );
+        index = std::stoi( index_as_string );
+    }
+
+    return index;
+}
+
+
+void dds_device_proxy::set_profile_intrinsics( std::shared_ptr< stream_profile_interface > & profile,
+                                               const std::shared_ptr< realdds::dds_stream > & stream ) const
+{
+    auto video_stream = std::dynamic_pointer_cast< realdds::dds_video_stream >( stream );
+    auto motion_stream = std::dynamic_pointer_cast< realdds::dds_motion_stream >( stream );
+    if( video_stream )
+    {
+        set_video_profile_intrinsics( profile, video_stream );
+    }
+    if( motion_stream )
```

Comment:
Can't be both... use `else` and inline the variable decl?

Reply:
Done